smarts/core/sumo_road_network.py

```diff
@@ -230,7 +230,10 @@ def _compute_road_polygons(self):
 
                 lane_to_poly[lane.getID()] = shape
 
+        self._snap_internal_holes(lane_to_poly)
+        self._snap_external_holes(lane_to_poly)
```

Comment:
You could factor the following out of the 2 functions you're calling here, like:
```python
    for lane_id in lane_to_poly:
        lane = self.lane_by_id(lane_id)
        if lane.getEdge().isSpecial():
            self._snap_internal_holes(lane)
        else:
            self._snaap_external_holes(lane)
```

(although, you may end up not wanting to do this if you change to a pattern for updating the dictionary outside of the loop, as suggested below.)

Reply:
It seems like this will work only when the internal edges are snapped first and then the external edges later. So I will have to iterate over the two edges separately.